clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -1040,39 +1040,49 @@ class ArchiveFileHandler final : public FileHandler {
     for (auto &C : Ar->children(Err)) {
       ++ChildIndex;
       auto BinOrErr = C.getAsBinary();
+
+      std::unique_ptr<FileHandler> FH{nullptr};
+      std::unique_ptr<MemoryBuffer> Buf{nullptr};
+
       if (!BinOrErr) {
         if (auto Err = isNotObjectErrorInvalidFileType(BinOrErr.takeError()))
           return Err;
-        continue;
-      }
 
-      auto &Bin = BinOrErr.get();
-      if (!Bin->isObject())
-        continue;
+        /* Handle bundled BC Files */
+        FH = std::make_unique<BinaryFileHandler>(BundlerConfig);
+        auto MR = C.getMemoryBufferRef();
+        assert(MR);
+        Buf = MemoryBuffer::getMemBuffer(*MR, false);
+      } else {
+        auto &Bin = BinOrErr.get();
+        if (!Bin->isObject())
+          continue;
 
-      auto CheckOrErr = CheckIfObjectFileContainsExcludedTargets(C);
-      if (!CheckOrErr)
-        return CheckOrErr.takeError();
+        auto CheckOrErr = CheckIfObjectFileContainsExcludedTargets(C);
```

Comment:
Hi @maksimsab 

We are trying to add support here for cases where the archive file will contain .bc file. Previously, we handle only those archive files which have .o files. 
Can you please comment if we will need this check for archives containing .bc files?

Thanks